tests/test_geopandas_provider.py

```diff
@@ -0,0 +1,268 @@
+    assert results['features'][0]['properties']['value'] == 89.9
+
+    assert results['features'][0]['geometry']['coordinates'][0] == -75.0
+    assert results['features'][0]['geometry']['coordinates'][1] == 45.0
+
+    results = p.query(limit=1)
+    assert len(results['features']) == 1
+    assert results['features'][0]['id'] == '371'
+
+    results = p.query(offset=2, limit=1)
+    assert len(results['features']) == 1
+    assert results['features'][0]['id'] == '238'
+    # should just be stn_id, datetime and value
+    assert len(results['features'][0]['properties']) == 3
+
+    results = p.query(select_properties=['value'])
+    assert len(results['features'][0]['properties']) == 1
+
+    results = p.query(select_properties=['value', 'stn_id'])
+    assert len(results['features'][0]['properties']) == 2
+
+    results = p.query(skip_geometry=True)
+    assert results['features'][0]['geometry'] is None
+
+    results = p.query(properties=[('stn_id', '604')])
+    assert len(results['features']) == 1
+    assert results['numberMatched'] == 1
+    assert results['numberReturned'] == 1
+
+    results = p.query(properties=[('stn_id', '35')])
+    assert len(results['features']) == 2
+    assert results['numberMatched'] == 2
+    assert results['numberReturned'] == 2
+
+    results = p.query(properties=[('stn_id', '35'), ('value', '93.9')])
+    assert len(results['features']) == 1
+
+    config['properties'] = ['value', 'stn_id']
+    p = GeoPandasProvider(config)
+    results = p.query()
+    assert len(results['features'][0]['properties']) == 2
+
+
+def test_csv_get(config):
+    p = GeoPandasProvider(config)
+
+    result = p.get('964')
+    assert result['id'] == '964'
+    assert result['properties']['value'] == 99.9
+
+
+def test_csv_get_not_existing_item_raise_exception(config):
+    """Testing query for a not existing object"""
+    p = GeoPandasProvider(config)
+    with pytest.raises(ProviderItemNotFoundError):
+        p.get('404')
+
+
+@pytest.fixture()
+def station_config():
+    return {
+        'name': 'CSV',
+        'type': 'feature',
+        'data': "data/station_list.csv",
+        'id_field': 'wigos_station_identifier',
+        'geometry': {
+            'x_field': 'longitude',
+            'y_field': 'latitude'
+        }
+    }
+
+def test_csv_get_station(station_config):
+    p = GeoPandasProvider(station_config)
+
+    results = p.query(limit=20)
+    assert len(results['features']) == 20
+    assert results['numberMatched'] == 79
+    assert results['numberReturned'] == 20
+
+    result = p.get('0-20000-0-16337')
+    assert result['properties']['station_name'] == 'BONIFATI (16337-0)'
+
+    result = p.get('0-454-2-AWSNAMITAMBO')
+    assert result['properties']['station_name'] == 'NAMITAMBO'
+
+
+@pytest.fixture()
+def gpkg_config():
+    return {
+        'name': 'gpkg',
+        'type': 'feature',
+        'data': "data/hu02.gpkg",
+        'id_field': 'HUC2',
+    }
```

Comment:
Declare all the configs at the top of the file. You should also be able to use the remote URL for this file.

```python
    'data': 'https://www.hydroshare.org/resource/3295a17b4cc24d34bd6a5c5aaf753c50/data/contents/hu02.gpkg'
```

Could reduce the number of files needed to run the tests by referencing stable remote sources.

Reply:
Can we keep this locally? The dataset is not super large but at the same time it significantly slows down test driven development to need to fetch it every time the environment is spun up. In general I also think that unit tests should isolate environment predictably and not fetch external resources. Even if the resource is stable I think it is stablest to have it in the repo and then remove it during packaging if we don't want to ship with it.

I can't change the pypi code to remove it before publishing